tests/test_rtdc_feat_temp.py

```diff
@@ -99,23 +99,78 @@ def test_filtering():
         dclab.register_temporary_feature("my_special_feature")
-        with pytest.raises(NotImplementedError):
-            dclab.set_temporary_feature(rtdc_ds=child,
-                                        feature="my_special_feature",
-                                        data=np.arange(len(child)))
+
+        dclab.set_temporary_feature(rtdc_ds=child,
+                                    feature="my_special_feature",
+                                    data=np.arange(len(child)))
+        assert "my_special_feature" in child.features
+        np.testing.assert_array_equal(child["my_special_feature"][:],
+                                      [0, 1, 2, 3, 4, 5, 6])
+
+
+@pytest.mark.filterwarnings(
+    "ignore::dclab.rtdc_dataset.config.WrongConfigurationTypeWarning")
+def test_hierarchy_children_parent_filtered():
+    """Test for RTDCHierarchy"""
+    h5path = retrieve_data("fmt-hdf5_fl_2018.zip")
+    with dclab.new_dataset(h5path) as ds:
+        ds.config["filtering"]["area_um min"] = 0.0
+        ds.config["filtering"]["area_um max"] = 33
+        ds.apply_filter()
+        child = dclab.new_dataset(ds)
+        dclab.register_temporary_feature("my_special_feature")
+
+        dclab.set_temporary_feature(rtdc_ds=child,
+                                    feature="my_special_feature",
+                                    data=np.arange(len(child)))
+        assert "my_special_feature" in child.features
+        np.testing.assert_array_equal(child["my_special_feature"][:],
+                                      [0, 1, 2, 3, 4, 5])
+        np.testing.assert_array_equal(ds["my_special_feature"][:],
+                                      [0, np.nan, 1, 2, 3, 4, 5])
+
+
+@pytest.mark.filterwarnings(
+    "ignore::dclab.rtdc_dataset.config.WrongConfigurationTypeWarning")
+def test_hierarchy_children_grandchildren():
+    """Test for RTDCHierarchy"""
+    h5path = retrieve_data("fmt-hdf5_fl_2018.zip")
+    with dclab.new_dataset(h5path) as ds:
+        ds.config["filtering"]["area_um min"] = 0.0
+        ds.config["filtering"]["area_um max"] = 33
+        ds.apply_filter()
+        child = dclab.new_dataset(ds)
+        child.config["filtering"]["deform min"] = 0
+        child.config["filtering"]["deform max"] = 0.012
+        child.rejuvenate()
+        grandchild = dclab.new_dataset(child)
+        dclab.register_temporary_feature("my_special_feature")
+
+        dclab.set_temporary_feature(rtdc_ds=child,
+                                    feature="my_special_feature",
+                                    data=np.arange(len(child)))
+        assert "my_special_feature" in child.features
+        np.testing.assert_array_equal(child["my_special_feature"][:],
+                                      [0, 1, 2, 3, 4, 5])
+        np.testing.assert_array_equal(ds["my_special_feature"][:],
+                                      [0, np.nan, 1, 2, 3, 4, 5])
+        assert "my_special_feature" in grandchild.features
```

Comment:
For the sake of completeness, please also add a `np.testing.assert_equal` for "my_special_feature" for the grandchild here.

Reply:
👍